api/src/pcapi/core/api_token/api_token.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime
+from datetime import timedelta
+import enum
+
+from flask import current_app as app
+
+from pcapi.core.users.exceptions import InvalidToken
+from pcapi.core.users.utils import decode_jwt_token
+from pcapi.core.users.utils import encode_jwt_payload
+
+
+class TokenType(enum.Enum):
+    TEST_TOKEN_TYPE_1 = "test1"  # this type is used in the tests only
+    TEST_TOKEN_TYPE_2 = "test2"  # this type is used in the tests only
+
+
+class ApiToken:
+    token_type: TokenType
+    user_id: str
+    encoded_token: str
+    data: dict
+
+    def __init__(self, encoded_token: str):
+        self.encoded_token = encoded_token
+        try:
+            payload = decode_jwt_token(encoded_token)
+            self.data = payload["data"] if payload["data"] is not None else {}
+            self.token_type = TokenType(payload["token_type"])
+            self.user_id = payload["user_id"]
+        except Exception:
+            raise InvalidToken()
+
+    def get_data(self) -> dict:
+        return self.data
+
+    def get_user_id(self) -> str:
+        return self.user_id
+
+    def get_token_type(self) -> TokenType:
+        return self.token_type
+
+    def get_encoded_token(self) -> str:
+        return self.encoded_token
+
+    def get_expiration_date(self) -> datetime | None:
+        ttl = app.redis_client.ttl(self.encoded_token)  # type: ignore [attr-defined]
+        if ttl < 0:
+            # -2 if doesn't exist, -1 if no expiration
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Yes ! Ce commentaire est top, il apporte de la comprhension